torchao/dtypes/aqt.py

```diff
@@ -140,9 +348,7 @@ def __init__(
         dtype=None,
         strides=None,
     ):
-        self.int_data = int_data
-        self.scale = scale
-        self.zero_point = zero_point
+        self.layout_tensor = layout_tensor
```

Comment:
I understand the benefit of group the above 3 configs into the same object but I don't think I'm quite clear on why this needs to be a subclass?

Reply:
we need a subclass so that we can support different packing format, like the ones that's used by tensor core

Reply:
With a non-standard layout, I don't doubt the need for a subclass (since you need to override the behavior of operations to know how to deal with the layout), but I doubt the need for a wrapper subclass, just do a basic storage and have all the smarts in the tensor. That's how non-contiguous layouts work on dense tensor!

Reply:
so I'd expect we do explicit conversion between different packing formats:
```
# plain layout
tensor = to_aq(float_tensor, ...)

tensor = tensor.cuda()
# for tensor core
tensor = tensor.to(extended_layout = "tensor_core_tiled")

# repack for cpu
# unpacking
tensor = tensor.to(extended_layout = "plain")
tensor = tensor.cpu()
tensor = tensor.to(extended_layout = "avx512")
```

Reply:
> the problem here is that we still want the same tensor (AffineQuantizedTensor) but can be packed into different formats (different storage)

I think my trouble is I didn't understand what you meant here. To give an analogy for regular PyTorch, if I have a 2D tensor, I can pack it into two different formats col-major, row-major, but they're usually represented as two separate tensors (they don't even share storage). So what do you mean "still want the same tensor"?

Reply:
oh sorry, I meant we still use the same tensor subclass, not the same tensor instance

Reply:
@ezyang - the analogy here is changing from strided to COO to BSR to strided etc.

Reply:
This will also be useful for https://github.com/pytorch/ao/blob/4c1d568263582db3e48299d2eba78d8ad1f6274d/torchao/dtypes/uint4.py to implement different bitpacking formats (different layouts).

Reply:
There are many ways to skin this cat. But the way I would think about it is that if you want a bunch of different behaviors to live in a single class (no extra subclass hierarchy), well, you can't use the class dispatch mechanism to dispatch them. So you need your own dispatch mechanism. Luckily, in Python, that's pretty easy to build.

Reply:
Yes, agreed. Coming up with a good way to skin this cat could help us make it a generic recipe for PyTorch users. @jerryzh168 and I have been chatting a bit about it.